superdesk/publish/http_push.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014, 2015 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import json
+
+from superdesk import config, app
+from superdesk.errors import PublishHTTPPushError
+from superdesk.publish import register_transmitter
+from superdesk.publish.publish_service import PublishService
+from io import BytesIO
+from bson import ObjectId
+from urllib.parse import urlparse
+from http.client import HTTPConnection, HTTPSConnection, InvalidURL
+from copy import copy
+
+
+errors = [PublishHTTPPushError.httpPushError().get_error_description()]
+
+
+class ItemNotFound(Exception):
+    pass
+
+class IdConflict(Exception):
+    pass
+
+
+class HTTPPushService(PublishService):
+    """HTTP Publish Service."""
+    headers = {"Content-type": "application/json", "Accept": "application/json"}
+
+    def _transmit(self, queue_item, subscriber):
+        config = queue_item.get('destination', {}).get('config', {})
+
+        try:
+            self._copy_published_media_files(json.loads(queue_item['formatted_item']))
+
+            try:
+                self._post(config.get('resource_url'), queue_item['formatted_item'])
```

Comment:
It's better to post all the time, I don't think we should fetch the existing story from content api and try to patch it. This sort of checks should be in Content API.